pkg/channel/consolidated/reconciler/controller/kafkachannel.go

```diff
@@ -530,15 +590,32 @@ func (r *Reconciler) updateKafkaConfig(ctx context.Context, configMap *corev1.Co
 	// Eventually the previous config should be snapshotted to delete Kafka topics
 	r.kafkaConfig = kafkaConfig
 	r.kafkaConfigError = err
+	ac, err := kafka.NewAdminClient(ctx, func() (sarama.ClusterAdmin, error) {
+		return source.MakeAdminClient(controllerAgentName, r.kafkaAuthConfig, kafkaConfig.Brokers)
+	})
+
+	if err != nil {
+		logger.Errorw("Error creating AdminClient", zap.Error(err))
+		return
+	}
+
+	if r.consumerGroupWatcher != nil {
+		r.consumerGroupWatcher.Terminate()
+	}
+
+	r.consumerGroupWatcher = NewConsumerGroupWatcher(ctx, ac, pollInterval)
+	//TODO handle error
+	r.consumerGroupWatcher.Start()
 }
 
 func (r *Reconciler) FinalizeKind(ctx context.Context, kc *v1beta1.KafkaChannel) pkgreconciler.Event {
 	// Do not attempt retrying creating the client because it might be a permanent error
 	// in which case the finalizer will never get removed.
-	if kafkaClusterAdmin, err := r.createClient(ctx, kc); err == nil && r.kafkaConfig != nil {
+	if kafkaClusterAdmin, err := r.createClient(); err == nil && r.kafkaConfig != nil {
 		if err := r.deleteTopic(ctx, kc, kafkaClusterAdmin); err != nil {
 			return err
 		}
 	}
+	r.consumerGroupWatcher.Forget(string(kc.ObjectMeta.UID))
 	return newReconciledNormal(kc.Namespace, kc.Name) //ok to remove finalizer
 }
```

Comment:
This won't be called on non-leader replicas, so to properly clean up resources we have to implement [`ReadOnlyFinalizer`](https://github.com/knative/pkg/blob/a2394e88959df2ca1900855474ae43d7da0a9046/codegen/cmd/injection-gen/generators/reconciler_reconciler.go#L255) or keep HA disabled and have only `ReconcileKind` and `FinalizeKind`. 
I think the latter solution is better (less code, no wasted resource to acquire locks, etc).

Reply:
yeah after some thinking of @lionelvillard comment, I think so too

Reply:
@pierDipi sorry I got a little confused, this is the controller not the dispatcher. The controller runs in HA with a single leader, and stand-by replicas. We don't need this called on non-leader replicas cause they don't set watchers first place.

Reply:
I was confused too :)

But I think the comment is valid (in the wrong place) since the dispatcher doesn't implement `Finalizer` at all, and that's a bug that we can address later.

Reply:
Opened an issue: https://github.com/knative-sandbox/eventing-kafka/issues/245